### PR TITLE
Update: [AEA-3859] - Create Apigee proxy for PSU API 

### PR DESCRIPTION
**Pull request by kris-szlapa**
## Summary

🎫 [AEA-3859](https://nhsd-jira.digital.nhs.uk/browse/AEA-3859) Create Apigee proxy for PSU API

- :exclamation: Breaking Change
- :robot: Operational or Infrastructure Change
- :sparkles: New Feature
- :warning: Potential issues that might be caused by this change

### Details

This pull request encompasses the creation of the Apigee proxy for the PSU API. This should be executed using Proxygen, employing an application-restricted authentication model, and interfacing with the PSU AWS API.

### CODE REVIEW

File: .github/scripts/deploy_api.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# proxygen_public_key_arn=$(aws cloudformation list-exports --query "Exports[?Name=='account-resources:ProxgenPublicKey'].Value" --output text)
+
+proxygen_private_key=$(aws secretsmanager get-secret-value --secret-id "${proxygen_private_key_arn}" --query SecretString --output text)
+# proxygen_public_key=$(aws secretsmanager get-secret-value --secret-id "${proxygen_public_key_arn}" --query SecretString --output text)
+
+# client_private_key_arn=$(aws cloudformation list-exports --query "Exports[?Name=='account-resources:PsuClientKeySecret'].Value" --output text)
+# client_public_key_arn=$(aws cloudformation list-exports --query "Exports[?Name=='account-resources:PsuClientCertSecret'].Value" --output text)
+
+# client_private_key=$(aws secretsmanager get-secret-value --secret-id "${client_private_key_arn}" --query SecretString --output text)
+# client_public_key=$(aws secretsmanager get-secret-value --secret-id "${client_public_key_arn}" --query SecretString --output text)
```
Comment (anthony-nhs) [inline .github/scripts/deploy_api.sh:29]: can you remove commented out code if its not used

Comment (kris-szlapa) [reply .github/scripts/deploy_api.sh]: Commented out code has been removed.

Comment (anthony-nhs) [reply .github/scripts/deploy_api.sh]: I have realized why they are there - we need some commands further down before we do a deploy to set the secrets for mutual tls using proxygen

File: .github/scripts/deploy_api.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+if [ "$TARGET_ENVIRONMENT" != "dev-pr" ]; then
+  environment=$TARGET_ENVIRONMENT
+else
+  environment=internal-dev
+fi
+
+echo "Environment name: $environment"
+
+# Find and replace the x-nhsd-apim.target.url value
+jq --arg stack_name "$STACK_NAME" '.["x-nhsd-apim"].target.url = "https://\($stack_name).dev.eps.national.nhs.uk"' "$SPEC_PATH" > temp.json && mv temp.json "$SPEC_PATH"
```
[the remark this answers is not in the document]
Comment (kris-szlapa) [reply .github/scripts/deploy_api.sh:17]: I'm uncertain about the values we receive from `$TARGET_ENVIRONMENT`. 
Are they dev-pr, qa, ref, int, or prod? And should these values be used to configure AWS environments?

[the remark this answers is not in the document]
Comment (kris-szlapa) [reply .github/scripts/deploy_api.sh]: `aws_environment=$(echo "$TARGET_ENVIRONMENT" | cut -d'-' -f1)`
```
jq --arg stack_name "$STACK_NAME" --arg aws_env "$aws_environment" '.["x-nhsd-apim"].target.url = "https://\($stack_name).\($aws_env).eps.national.nhs.uk"' "$SPEC_PATH" > temp.json && mv temp.json "$SPEC_PATH"
```

File: .github/scripts/deploy_api.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# proxygen_public_key=$(aws secretsmanager get-secret-value --secret-id "${proxygen_public_key_arn}" --query SecretString --output text)
+
+# client_private_key_arn=$(aws cloudformation list-exports --query "Exports[?Name=='account-resources:PsuClientKeySecret'].Value" --output text)
+# client_public_key_arn=$(aws cloudformation list-exports --query "Exports[?Name=='account-resources:PsuClientCertSecret'].Value" --output text)
+
+# client_private_key=$(aws secretsmanager get-secret-value --secret-id "${client_private_key_arn}" --query SecretString --output text)
+# client_public_key=$(aws secretsmanager get-secret-value --secret-id "${client_public_key_arn}" --query SecretString --output text)
+
+# Save private keys to temporary files
+echo "$proxygen_private_key" > /tmp/proxygen_private_key.pem
```
[the remark this answers is not in the document]
Comment (kris-szlapa) [reply .github/scripts/deploy_api.sh]: Pivate key directory has been changed:
`echo "$proxygen_private_key" > ~/.proxygen/tmp/proxygen_private_key.pem`
`private_key_path: tmp/proxygen_private_key.pem`

File: .github/scripts/deploy_api.sh (hunk 1 above)
Comment (kris-szlapa) [reply .github/scripts/deploy_api.sh]: Putting secret psu-mtls-2 in internal-dev
```
{
  "name": "psu-mtls-2",
  "last_modified": "2024-04-16T14:11:10.420000+00:00",
  "environment": "internal-dev",
  "version_id": "be3c5980-5bce-40c6-a360-445c3107cb0a",
  "type": "mtls",
  "in_use_by": [
    "prescription-status-update-api--internal-dev--psu-pr-99-sandbox"
  ]
}
```

File: .github/scripts/deploy_api.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (kris-szlapa) [reply .github/scripts/deploy_api.sh]: Extract the AWS environment name from the target environment
```
aws_environment=$(echo "$TARGET_ENVIRONMENT" | cut -d'-' -f1)
echo "AWS environment: $aws_environment"
```
Determine the proxy environment based on the target environment
```
case "$TARGET_ENVIRONMENT" in
    dev-pr|dev)
        environment=internal-dev
        ;;
    qa)
        environment=internal-qa
        ;;
    *)
        environment=$TARGET_ENVIRONMENT
        ;;
esac
echo "Proxy environment: $environment"
```

**Review by anthony-nhs — APPROVED** [review]

**Pull request merged**